app/javascript/dashboard/helper/auditlogHelper.js

```diff
@@ -0,0 +1,140 @@
+const roleMapping = {
+  0: 'agent',
+  1: 'administrator',
+};
+
+const availabilityMapping = {
+  0: 'online',
+  1: 'offline',
+  2: 'busy',
+};
+
+const translationKeys = {
+  'automationrule:create': `AUDIT_LOGS.AUTOMATION_RULE.ADD`,
+  'automationrule:update': `AUDIT_LOGS.AUTOMATION_RULE.EDIT`,
+  'automationrule:destroy': `AUDIT_LOGS.AUTOMATION_RULE.DELETE`,
+  'webhook:create': `AUDIT_LOGS.WEBHOOK.ADD`,
+  'webhook:update': `AUDIT_LOGS.WEBHOOK.EDIT`,
+  'webhook:destroy': `AUDIT_LOGS.WEBHOOK.DELETE`,
+  'inbox:create': `AUDIT_LOGS.INBOX.ADD`,
+  'inbox:update': `AUDIT_LOGS.INBOX.EDIT`,
+  'inbox:destroy': `AUDIT_LOGS.INBOX.DELETE`,
+  'user:sign_in': `AUDIT_LOGS.USER_ACTION.SIGN_IN`,
+  'user:sign_out': `AUDIT_LOGS.USER_ACTION.SIGN_OUT`,
+  'team:create': `AUDIT_LOGS.TEAM.ADD`,
+  'team:update': `AUDIT_LOGS.TEAM.EDIT`,
+  'team:destroy': `AUDIT_LOGS.TEAM.DELETE`,
+  'macro:create': `AUDIT_LOGS.MACRO.ADD`,
+  'macro:update': `AUDIT_LOGS.MACRO.EDIT`,
+  'macro:destroy': `AUDIT_LOGS.MACRO.DELETE`,
+  'accountuser:create': `AUDIT_LOGS.ACCOUNT_USER.ADD`,
+  'accountuser:update:self': `AUDIT_LOGS.ACCOUNT_USER.EDIT.SELF`,
+  'accountuser:update:other': `AUDIT_LOGS.ACCOUNT_USER.EDIT.OTHER`,
+};
+
+function extractAttrChange(attrChange) {
+  if (Array.isArray(attrChange)) {
+    return attrChange[attrChange.length - 1];
+  }
+  return attrChange;
+}
+
+export function extractChangedAccountUserValues(audited_changes) {
+  let changes = [];
+  let values = [];
+
+  // Check roles
+  if (audited_changes.role && audited_changes.role.length) {
+    changes.push('role');
+    values.push(roleMapping[extractAttrChange(audited_changes.role)]);
+  }
+
+  // Check availability
+  if (audited_changes.availability && audited_changes.availability.length) {
+    changes.push('availability');
+    values.push(
+      availabilityMapping[extractAttrChange(audited_changes.availability)]
+    );
+  }
+
+  return { changes, values };
+}
+
+function handleAccountUserCreate(
+  auditLogItem,
+  translationPayload,
+  getAgentName
```

Comment:
Passing a function like this is generally considered an anti-pattern, especially the function is Vue specific. We can move this helper to a mixin instead, let me know if you need help in refactoring. 

I think we can move the `handleAccountUserCreate`, `handleAccountUserUpdate` and `generateTranslationPayload` methods to the component itself, or a mixin if there's scope of reusability